connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaOffsetBackingStore.java

```diff
@@ -118,8 +118,11 @@ public void stop() {
     public Future<Void> set(final Map<ByteBuffer, ByteBuffer> values, final Callback<Void> callback) {
         SetCallbackFuture producerCallback = new SetCallbackFuture(values.size(), callback);
 
-        for (Map.Entry<ByteBuffer, ByteBuffer> entry : values.entrySet())
-            offsetLog.send(entry.getKey().array(), entry.getValue().array(), producerCallback);
+        for (Map.Entry<ByteBuffer, ByteBuffer> entry : values.entrySet()) {
+            ByteBuffer key = entry.getKey();
+            ByteBuffer value = entry.getValue();
+            offsetLog.send(key == null ? null : key.array(), value == null ? null : value.array(), producerCallback);
```

Comment:
The `null` for key is really for completeness, although admittedly in practice we shouldn't ever hit it because of namespacing. For namespacing, check `OffsetStorageWriter` and `OffsetStorageReader`, the namespacing is implemented at that level for keys.